### PR TITLE
Animations: Effect Chooser Component

### PR DESCRIPTION
**Pull request by carloskelly13**
## Summary

Adds the picker control to be used in the timeline and dropdown for the first five animations. 

Screen Recording 2020-11-02 at 5 08 03 PM-Animated Image (Large)

## Relevant Technical Choices

- Uses CSS Keyframes for animations.
- No providers to keep performance high and 60fps

## To-do

- Add the rest of the animations and a callback

## User-facing changes

None

## Testing Instructions

1. Visit Storybook
2. View Animations > Effect Chooser
3. See the animations loop

---

#3804

### CODE REVIEW

File: assets/src/edit-story/components/panels/animation/effectChooser.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+/**
+ * Internal dependencies
+ */
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import loadStylesheet from '../../../utils/loadStylesheet';
+import { GOOGLE_MENU_FONT_URL } from '../../../app/font';
```
Comment (maxyinger) [inline assets/src/edit-story/components/panels/animation/effectChooser.js]: fix import comments

File: assets/src/edit-story/components/panels/animation/effectChooserElements.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    animation-timing-function: cubic-bezier(.22,.61,.35,1);
+  }
+  41.5% {
+    transform: translateY(-30%);
+    animation-timing-function: cubic-bezier(.75,.05,.86,.08);
+  }
+  50%, 100% {
+    transform: translateY(0%);
+    animation-timing-function: cubic-bezier(.22,.61,.35,1);
+  }
```
Comment (maxyinger) [inline assets/src/edit-story/components/panels/animation/effectChooserElements.js:48]: nice easy delay

File: assets/src/edit-story/components/panels/animation/effectChooserElements.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  50%, 100% {
+    transform: translateX(0%);
+  }
+`;
+
+export const RotateInRightAnimation = styled.div`
+  display: inline-block;
+  animation: ${rotateInRightKeyframes} 2s linear infinite;
+`;
```
Comment (maxyinger) [inline assets/src/edit-story/components/panels/animation/effectChooserElements.js:172]: The simplified versions of the anims as css animations look great! Also makes it super easy to only apply them on hover if we want. nice work!

**Review by maxyinger — APPROVED** [review]
awesome work! The css anims look great and will be super nice to alter or add new ones. 🍻

File: assets/src/edit-story/components/panels/animation/effectChooser.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  font-family: 'Teko', sans-serif;
+  font-size: 20px;
+  color: white;
+  text-transform: uppercase;
+`;
+
+export default function EffectChooser() {
+  useEffect(() => {
+    loadStylesheet(`${GOOGLE_MENU_FONT_URL}?family=Teko`).catch(function () {});
+  });
```
Comment (maxyinger) [inline assets/src/edit-story/components/panels/animation/effectChooser.js:92]: Would using:
```
const {
    actions: { maybeEnqueueFontStyle },
  } = useFont();
```
prevent double loading this if one of the text sets or stories is already loading in Teko?

Or is this doing something that `maybeEnqueueFontStyle` doesn't do?

File: assets/src/edit-story/components/panels/animation/effectChooser.js (hunk 1 above)
Comment (carloskelly13) [reply assets/src/edit-story/components/panels/animation/effectChooser.js]: doh! This is the edge case where Eslint fix can sneak by a false positive! Thank you!

File: assets/src/edit-story/components/panels/animation/effectChooser.js (hunk 4 above)
Comment (carloskelly13) [reply assets/src/edit-story/components/panels/animation/effectChooser.js:92]: `maybeEnqueueFontStyle` downloads only the chars we need for the font menu. Ideally we might make Teko a required app font.

Comment (maxyinger) [reply assets/src/edit-story/components/panels/animation/effectChooser.js:92]: Ah makes sense. thanks for clarifying!

**Pull request merged**